test/key.test.ts

```diff
@@ -1,100 +1,40 @@
 import {
-    generatePrivateKey,
-    getPublicFromPrivate,
-    recoverEcdsa,
-    recoverSchnorr,
-    SchnorrSignature,
-    signEcdsa,
-    signSchnorr,
-    verifyEcdsa,
-    verifySchnorr
+    Key,
+    // recoverEcdsa,
+    // recoverSchnorr,
+    // SchnorrSignature,
+    // signEd25519,
+    // signSchnorr,
+    // verifyEd25519,
+    // verifySchnorr
+    Signature
 } from "..";
 
-test("generatePrivateKey", () => {
-    const priv = generatePrivateKey();
-    expect(/^[0-9a-fA-F]{64}$/.test(priv)).toBe(true);
+test("generatePrivateKey", async () => {
+    const key = new Key();
+    await key.init();
+    const priv = key.generatePrivateKey();
+    expect(/^[0-9a-fA-F]{128}$/.test(priv)).toBe(true);
 });
 
-test("getPublicFromPrivate", () => {
-    const priv = generatePrivateKey();
-    const pub = getPublicFromPrivate(priv);
-    expect(/^[0-9a-fA-F]{128}$/.test(pub)).toBe(true);
+test("getPublicFromPrivate", async () => {
+    const key = new Key();
+    await key.init();
+    const priv = key.generatePrivateKey();
+    const pub = key.getPublicFromPrivate(priv);
+    expect(/^[0-9a-fA-F]{64}$/.test(pub)).toBe(true);
 });
 
-test("sign & verify ECDSA", () => {
+test("sign & verify EdDSA", async () => {
```

Comment:
Let's use more specific name for the signature scheme, "Ed25519"

Reply:
I recommend you to generate testing oracles from other implementation like the previous.